docs/rfcs/1011-non-blocking-invite-transport.md

```diff
@@ -0,0 +1,1238 @@
+#### `invite_claimer_cancel_attempt` command
+
+This is the new invited command introduced for the claimer to cancel an attempt with a provided reason.
+
+In order for the claimer to cancel an attempt, a couple of checks have to be performed:
+- the greeter exists
+- the greeter is allowed (i.e. it is part of the greeters)
+- the attempt exists
+- the attempt is active (i.e. joined and not cancelled)
+
+> [!NOTE]
+> The state of the invitation doesn't have to be checked since it is already checked by the `/invited` HTTP route.
+
+Otherwise this command has the same logic as `invite_greeter_cancel_attempt`.
+
+Pseudo-code:
+
+```python
+def invite_claimer_cancel_attempt(
+    token: InvitationToken,
+    greeter: GreeterID,
+    attempt: AttemptID,
+    reason: CancelledAttemptReason,
+) ->
+  InvitationNotFound | InvitationCompleted | InvitationCancelled
+  AttemptNotFound | AttemptNotJoined | AttemptCancelled
+  GreeterNotFound | GreeterNotAllowed | OK:
+    # Perform checks
+    [...]
+    # Call `claimer_cancel_attempt(reason)`
+    [...]
+```
+
+Schema definition:
+
+```json5
+[
+    {
+        "major_versions": [
+            4
+        ],
+        "req": {
+            "cmd": "invite_claimer_cancel_attempt",
+            "fields": [
+                {
+                    "name": "token",
+                    "type": "InvitationToken"
+                },
+                {
+                    "name": "greeter",
+                    "type": "UserID"
+                },
+                {
+                    "name": "attempt",
+                    "type": "AttemptID"
+                },
+                {
+                    "name": "reason",
+                    "type": "CancelledAttemptReason"
+                }
+            ]
+        },
+        "reps": [
+            {
+                "status": "ok"
+            },
+            // The following statuses do exist for the greeter, but not for the claimer
+            // Instead, in those cases, the claimer would get an HTTP 410 error, defined
+            // as `InvitationAlreadyUsedOrDeleted`.
+            // {
+            //     // The invitation token doesn't correspond to any existing invitation
+            //     "status": "invitation_not_found"
+            // },
+            // {
+            //     // The invitation has already been completed
+            //     "status": "invitation_completed"
+            // },
+            // {
+            //     // The invitation has been cancelled
+            //     "status": "invitation_cancelled"
+            // },
+            {
+                // The provided greeter ID doesn't correspond to any existing greeter
+                "status": "greeter_not_found"
+            },
+            {
+                // The greeter is not part of the allowed greeters for this invitation
+                "status": "greeter_not_allowed"
+            },
+            {
+                // The attempt id doesn't correspond to any existing attempt
```

Comment:
```suggestion
                 // No attempt joined by the greeter corresponds to this attempt ID
```

(the change is here to show that using existing attempt ID that is unrelated to this greeter will end up with this error)

Reply:
Per @mmmarcos decision we keep those errors separated